examples/advanced/lr-newton-raphson/README.md

```diff
@@ -0,0 +1,243 @@
+
+Using `nvflare`, The federated logistic regression with Newton-Raphson
+optimization is implemented as follows.
+
+On the server side, all workflow logics are implemented in
+class `FedAvgNewtonRaphson`, which can be found
+[here](job/newton_raphson/app/custom/newton_raphson_workflow.py). The
+`FedAvgNewtonRaphson` class inherits from the
+[`BaseFedAvg`](https://github.com/NVIDIA/NVFlare/blob/main/nvflare/app_common/workflows/base_fedavg.py)
+class, which itself inherits from the **Workflow Controller**
+([`WFController`](https://github.com/NVIDIA/NVFlare/blob/main/nvflare/app_common/workflows/wf_controller.py))
+class. This is the preferrable approach to implement a custom
+workflow, since `WFController` decouples communication logic from
+actual workflow (training & validation) logic. The mandatory
+method to override in `WFController` is the
+[`run()`](https://github.com/NVIDIA/NVFlare/blob/main/nvflare/app_common/workflows/wf_controller.py#L37)
+method, where the orchestration of server-side workflow actually
+happens. The implementation of `run()` method in
+[`FedAvgNewtonRaphson`](job/newton_raphson/app/custom/newton_raphson_workflow.py)
+is similar to the classic
+[`FedAvg`](https://github.com/NVIDIA/NVFlare/blob/main/nvflare/app_common/workflows/fedavg.py#L44):
+- Initialize the global model, this is acheived through method `load_model()`
+  from base class
+  [`ModelController`](https://github.com/NVIDIA/NVFlare/blob/fa4d00f76848fe4eb356dcde417c136047eeab36/nvflare/app_common/workflows/model_controller.py#L292),
+  which relies on the
+  [`ModelPersistor`](https://nvflare.readthedocs.io/en/main/glossary.html#persistor). A
+  custom
+  [`NewtonRaphsonModelPersistor`](job/newton_raphson/app/custom/newton_raphson_persistor.py)
+  is implemented in this example, which is based on the
+  [`NPModelPersistor`](https://github.com/NVIDIA/NVFlare/blob/main/nvflare/app_common/np/np_model_persistor.py)
+  for numpy data, since the _model_ in the case of logistic regression
+  is just the parameter vector $\theta$ that can be represented by a
+  numpy array. Only the `__init__` method needs to be re-implemented
+  to provide a proper initialization for the global parameter vector
+  $\theta$.
+- During each training round, the global model will be sent to the
+  list of participating clients to perform a training task. This is
+  done using the
+  [`send_model()`](https://github.com/NVIDIA/NVFlare/blob/d6827bca96d332adb3402ceceb4b67e876146067/nvflare/app_common/workflows/model_controller.py#L99)
+  method under the hood, from the `ModelController` base class. Once
+  the clients finish their local training, results will be collected
+  and sent back to server as
+  [`FLModel`](https://nvflare.readthedocs.io/en/main/programming_guide/fl_model.html#flmodel)s.
+- Results sent by clients contain their locally computed gradient and
+  Hessian. A [custom aggregation
+  function](job/newton_raphson/app/custom/newton_raphson_workflow.py#L111)
```

Comment:
The link doesn't seem to work on github. For some reason it links to https://github.com/zhijinl/NVFlare/tree/lr-newton-raphson/examples/advanced/job/newton_raphson/app/custom/newton_raphson_workflow.py#L111, missing the `examples/advanced/lr-newton-raphson` folder. Maybe try using `./job/...`

Reply:
Tried using "./job/.." or removing the line number reference in the link: nothing works ...

Strange, because all other links work

Reply:
It doesn't seem to work with the line number. Best to remove #L111 for now.

Reply:
Removed line number.